### PR TITLE
Align naming: structure to parameters

### PR DESCRIPTION
**Pull request by jkrvivian**
* manaStructure -> manaParameters
* workScoreStructure -> workScoreParameters

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -112,8 +112,8 @@ type ProtocolParameters interface {
 	Bech32HRP() NetworkPrefix
 	// RentStructure defines the rent structure used by given node/network.
 	RentParameters() *RentParameters
-	// WorkScoreStructure defines the work score structure used by the given network.
-	WorkScoreStructure() *WorkScoreStructure
+	// WorkScoreParameters defines the work score structure used by the given network.
```
Comment (muXxer) [inline api.go]: please change the comment as well.

File: api_protocol_parameters.go [hunk 2]
```diff
@@ -11,8 +11,8 @@ type basicProtocolParameters struct {
 
 	// RentStructure defines the rent structure used by given node/network.
 	RentParameters RentParameters `serix:"3,mapKey=rentParameters"`
-	// WorkScoreStructure defines the work score structure used by given node/network.
-	WorkScoreStructure WorkScoreStructure `serix:"4,mapKey=workScoreStructure"`
+	// WorkScoreParameters defines the work score structure used by given node/network.
```
Comment (muXxer) [inline api_protocol_parameters.go]: please change the comment as well.

**Review by muXxer — CHANGES_REQUESTED** [review]

**Review by muXxer — APPROVED** [review]

**Comment by alexsporn** [comment]
Some of the objects (Blocks and Transactions) have now an API inside and could use that one to calculate the WorkScore.
Also the TransactionBuilder does not require having it as a parameter because it contains a valid API inside.

We can do these changes in a separate PR if you don't want to do them here.

**Comment by cyberphysic4l** [comment]
A further change we could make here is to do something similar to what we do with the StorageScoreParameter/StorageScoreStructure. The Parameters are what we set, and the Structure contains the Parameters plus some other values derived from the parameters, and this lives in the API. MaxBlockWorkScore, which is derived from the WorkScoreParameters would live inside this WorkScoreStructure instead of being on its own in the API in this case.

This way of doing things requires a little bit of bloating of the API but a nicer way to organise imo.

**Pull request closed without merging**